### DISCUSSION TITLE
Webpack v5/probe.gl; exports is not defined

### DISCUSSION THREAD

**Discussion by jfuehner** [visgl/deck.gl#6793] [Q&A] [answered]
Hi All!

Hoping for a little help. I just recently switched to using webpack v5.

After building my app for a production release I receive an “exports is not defined” error from within decks probe.gl dependency. This error prevents deck from loading.

The interesting thing is the error only occurs when I build for production. All works fine with development builds; no errors.

Any idea how I can fix this? I am currently running deck v8.6 but have tried with v8.7 and the issue still persist.

**Comment by ibgreen** [comment]
@jfuehner You are not providing much detail, probe.gl has a number of scoped submodules and the main `probe.gl`module is kind of deprecated. We  had issues with direct imports of a subfile `probe.gl/env`, this was replaced with a module `@probe.gl/env` in 8.7.

Lock files are tricky, some dependency, like luma.gl might hold on the older probe.gl files. You could try to clear the lock file and do a clean install and see if it makes a difference or provide more detailed into.

**Comment by jfuehner** [comment]
Hey @ibgreen! Hope all is well. Apologies for the somewhat lacking initial post. 

The last three lines of my stacktrace leading up to the exception are:
- @babel/runtime/helpers/esm/defineProperty.js
- @probe.gl/log/dist/esm/log.js
- @deck.gl/core/dist/esm/utils/log.js

I did completely delete my package-lock and node_modules to ensure everything was fresh. However, neither helped.

**Comment by jfuehner** [comment, marked as answer]
@ibgreen you can disregard this. Turns out there was a babel plugin that was missed during my upgrade to webpack 5. So, deck wasn't being transpiled correctly.
